File: 204-count-primes/count-primes.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n==0 and n==1:
            return 0
        count=0
        prime=[True]*(n)
        i=2
        while i*i<=n:
            if prime[i]==True:
                j=i*i
                while j<n:
                    prime[j]=False
                    j+=i
            i+=1

        for i in range(2,n):
            if prime[i]:
                count+=1
        return count

File: 204-count-primes/count-primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        int count=0;
        if(n==0 || n==1){
            return 0;
        }
        bool arr[n];
        for(int i=0;i<n;i++){
            arr[i]=true;
        }
        for(int i=2;i*i<n;i++){
            if(arr[i]==true){
                for(int j=i*i;j<n;j+=i){
                    arr[j]=false;
                }
            }
        }
        for(int i=2;i<n;i++){
            if(arr[i]==true){
                count++;
            }
        }
        return count;
    }

};